packages/jest-haste-map/src/index.js

```diff
@@ -638,6 +640,9 @@ class HasteMap extends EventEmitter {
           this._workerPromise = null;
           if (promise) {
             return promise.then(add);
+          } else {
+            // If a file in node_modules has changed, emit an event regardless.
+            add();
           }
         } else {
```

Comment:
Oh, got it. I prefer this to be more explicit because the haste map code is really complex.